product_warranty/models/product_supplierinfo.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2015 Vauxoo
+# Copyright (C) 2009-2011  Akretion, Emmanuel Samyn, Benoît Guillot      #
+#                                                                        #
+# This program is free software: you can redistribute it and/or modify   #
+# it under the terms of the GNU General Public License as published by   #
+# the Free Software Foundation, either version 3 of the License, or      #
+# (at your option) any later version.                                    #
+#                                                                        #
+# This program is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
+# GNU General Public License for more details.                           #
+#                                                                        #
+# You should have received a copy of the GNU General Public License      #
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.  #
+##########################################################################
+
+from openerp import api, fields, models
+
+
+class ProductSupplierInfo(models.Model):
+
+    _inherit = "product.supplierinfo"
+
+    @api.model
+    def get_warranty_return_partner(self):
+        result = [('company', 'Company'),
+                  ('supplier', 'Supplier'),
+                  ('other', 'Other'),
+                  ]
+        return result
+
+    @api.model
+    def _get_default_instructions(self):
+        """ Get selected lines to add to exchange """
+        instruction_ids = self.env['return.instruction']\
+            .search([('is_default', '=', True)], limit=1)
+        return instruction_ids
+
+    @api.one
+    @api.depends('warranty_return_partner')
+    def _get_warranty_return_address(self):
+        """ Method to return the partner delivery address or if none,
+        the default address
+
+        dedicated_delivery_address stand for the case a new type of
+        address more particularly dedicated to return delivery would be
+        implemented.
+
+        """
+        for record in self:
+            return_partner = record.warranty_return_partner
+            partner_id = record.company_id.partner_id.id
+            if return_partner:
+                if return_partner == 'supplier':
+                    partner_id = record.name.id
+                elif return_partner == 'company':
+                    if record.company_id.crm_return_address_id:
+                        partner_id = record.company_id.\
+                            crm_return_address_id.id
+                elif return_partner == 'other':
+                    if record.warranty_return_other_address:
+                        partner_id = record.\
+                            warranty_return_other_address.id
+                record.warranty_return_address = partner_id
```

Comment:
I think this should be unindented, at the moment it's inside the `if` block which I don't think is correct: we want to assign the return address to the company address if there is no return partner.
If that's not the case, then the line `partner_id = record.company_id.partner_id.id` is useless and can be removed.